### PR TITLE
Add new states to actors.

### PR DESCRIPTION
**Pull request by fmassot**
Two new states are now replacing the `Running` state:
- `Processing`: it corresponds to an actor that is processing messages, it has at least one message
- `Idle`: when an actor has no more message to process, it goes in `Idle` state.

## Purpose
It will be useful to define the right condition to terminate the indexing pipeline.

### CODE REVIEW

File: quickwit-actors/src/async_actor.rs [hunk 1]
```diff
@@ -131,13 +130,17 @@ async fn process_msg<A: Actor + AsyncActor>(
     }
 
     match command_or_msg_recv_res {
-        Ok(CommandOrMessage::Command(cmd)) => process_command(actor, cmd, ctx, state_tx),
+        Ok(CommandOrMessage::Command(cmd)) => {
+            ctx.start_processing();
+            process_command(actor, cmd, ctx, state_tx)
+        }
         Ok(CommandOrMessage::Message(msg)) => {
             let span = actor.message_span(msg_id, &msg);
```
Comment (fulmicoton) [inline quickwit-actors/src/async_actor.rs:138]: if your first message is message then I believe this is wrong.

File: quickwit-actors/src/sync_actor.rs [hunk 2]
```diff
@@ -114,13 +113,17 @@ fn process_msg<A: Actor + SyncActor>(
         return Some(ActorExitStatus::Killed);
     }
     match command_or_msg_recv_res {
-        Ok(CommandOrMessage::Command(cmd)) => process_command(actor, cmd, ctx, state_tx),
+        Ok(CommandOrMessage::Command(cmd)) => {
+            ctx.start_processing();
```
Comment (fulmicoton) [inline quickwit-actors/src/sync_actor.rs:117]: same as for async_actor.

File: quickwit-actors/src/actor_state.rs [hunk 3]
```diff
@@ -49,28 +51,54 @@ impl From<ActorState> for AtomicState {
...
+        *self == ActorState::Idle || *self == ActorState::Processing
+    }
+}
+
 pub struct AtomicState(AtomicU32);
 
 impl Default for AtomicState {
     fn default() -> Self {
-        AtomicState(AtomicU32::new(ActorState::Running as u32))
+        AtomicState(AtomicU32::new(ActorState::Processing as u32))
```
Comment (fulmicoton) [inline quickwit-actors/src/actor_state.rs:64]: Yes exactly!

File: quickwit-actors/src/actor_state.rs [hunk 4]
```diff
@@ -22,17 +22,19 @@ use std::sync::atomic::{AtomicU32, Ordering};
 #[repr(u32)]
 #[derive(Clone, Debug, Copy, Eq, PartialEq)]
 pub enum ActorState {
-    Running = 0,
-    Paused = 1,
-    Exit = 2,
+    Processing = 0,
```
Comment (fulmicoton) [inline quickwit-actors/src/actor_state.rs:25]:
```suggestion
    /// Processing implies that the actor has some message to process.
    Processing = 0,
    /// Idle means that the actor is currently waiting for messages.
```

**Review by fulmicoton — CHANGES_REQUESTED** [review]
I think you need to set running in more places.

File: quickwit-actors/src/async_actor.rs (hunk 1 above)
Comment (fmassot) [reply quickwit-actors/src/async_actor.rs:138]: oh yes totally wrong, I was not really awake when I committed this.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**